e2e/cluster.go

```diff
@@ -244,19 +273,31 @@ func hasViableConfig(scenario *scenario.Scenario, clusterConfigs []clusterConfig
 	return false
 }
 
-func createMissingClusters(
-	ctx context.Context,
-	r *mrand.Rand,
-	cloud *azureClient,
-	suiteConfig *suite.Config,
-	scenarios scenario.Table,
-	clusterConfigs *[]clusterConfig) error {
+func hasViableAirgapConfig(scenario *scenario.Scenario, clusterConfigs []clusterConfig) bool {
+	if !scenario.Airgap {
+		return true // config not needed for scenario
+	}
+	for _, config := range clusterConfigs {
+		if config.isAirgapCluster && scenario.Config.ClusterSelector(config.cluster) {
```

Comment:
ideally would like to keep things consistent with how we select on cluster properties - maybe we should refactor towards having cluster selectors/mutators work on cluster configs rather than individual cluster models

Reply:
also can't we just merge the airgap logic here into the existing `hasViableConfig`?  this seems like a lot of duplication

Reply:
The createMissingClusters logic makes things a bit confusing which is why I made the hasViableAirgapConfig - but let me take another look at it since I wrote this section last week when I was figuring a lot of stuff out

Reply:
fair enough, as I said before I'd like to do a lot of refactoring here some day :)